components/heap/heap_trace_standalone.c

```diff
@@ -239,47 +291,32 @@ static void heap_trace_dump_base(bool internal_ram, bool psram)
         printf("(NB: Internal Buffer has overflowed, so trace data is incomplete.)\n");
     }
     printf("================================\n");
+
+    portEXIT_CRITICAL(&trace_mux);
 }
 
 /* Add a new allocation to the heap trace records */
-static IRAM_ATTR void record_allocation(const heap_trace_record_t *record)
+static IRAM_ATTR void record_allocation(const heap_trace_record_t *rAllocation)
 {
-    if (!tracing || record->address == NULL) {
+    if (!tracing || rAllocation->address == NULL) {
         return;
     }
 
     portENTER_CRITICAL(&trace_mux);
 
     if (tracing) {
```

Comment:
Maybe this is not necessary since it is checked  on line 334. I missed it on the last PR.

Reply:
we must re-check while holding the mutex. the first check is a fast path.